lib/skate_web/auth_manager.ex

```diff
@@ -10,20 +10,51 @@ defmodule SkateWeb.AuthManager do
   @skate_admin_group "skate-admin"
   @skate_dispatcher_group "skate-dispatcher"
   @v2_resource_prefix "v2:"
+  @v3_resource_prefix "v3:"
 
   def v2_resource_prefix, do: @v2_resource_prefix
 
+  def v3_resource_prefix, do: @v3_resource_prefix
+
   def subject_for_token(%{id: user_id}, _claims) do
-    {:ok, "#{@v2_resource_prefix}#{user_id}"}
+    keycloak_enabled? =
+      "keycloak-sso" in Enum.map(Skate.Settings.TestGroup.get_override_enabled(), & &1.name)
+
+    if keycloak_enabled? do
+      {:ok, "#{@v3_resource_prefix}#{user_id}"}
+    else
+      {:ok, "#{@v2_resource_prefix}#{user_id}"}
+    end
   end
 
   def resource_from_claims(%{"sub" => @v2_resource_prefix <> user_id}) do
     {:ok, %{id: String.to_integer(user_id)}}
   end
 
+  def resource_from_claims(%{"sub" => @v3_resource_prefix <> user_id}) do
+    {:ok, %{id: String.to_integer(user_id)}}
+  end
+
   def resource_from_claims(_), do: {:error, :invalid_claims}
 
-  def username_from_socket!(socket) do
+  def verify_claims(%{"sub" => subject} = claims, _options) do
+    keycloak_enabled? =
+      "keycloak-sso" in Enum.map(Skate.Settings.TestGroup.get_override_enabled(), & &1.name)
+
+    if keycloak_enabled? do
+      case subject do
+        @v3_resource_prefix <> _user_ -> {:ok, claims}
+        _ -> {:error, :invalid_claims}
+      end
+    else
+      case subject do
+        @v2_resource_prefix <> _user_id -> {:ok, claims}
+        _ -> {:error, :invalid_claims}
+      end
+    end
```

Comment:
suggestion&nit(non-blocking): I like this syntax better (though I wish that `keycloak_enabled?` was a `auth_provider` variable that was `:cognito` or `:keycloak`) thoughts?
```suggestion
    case {keycloak_enabled?, subject} do
        {true, @v3_resource_prefix <> _user_id} -> {:ok, claims}
        {false, @v2_resource_prefix <> _user_id} -> {:ok, claims}
        _ -> {:error, :invalid_claims}        
    end
```

Reply:
I like this a lot better, and I think we can do the `auth_provider` thing! If you want to get credit, you can issue a larger suggestion that also changes lines 41-42 to instead do `auth_provider = if "keycloak-sso" in ...`.

Reply:
I could commit it directly if you'd like @lemald

Reply:
Works for me.

Reply:
7ae373ea283d366192e35450effb628f8884e267

Reply:
Excellent, LGTM.